cloudferrylib/os/network/neutron.py

```diff
@@ -805,21 +805,18 @@ def upload_networks(self, networks):
                     'name': net['name'],
                     'admin_state_up': net['admin_state_up'],
                     'tenant_id': tenant_id,
-                    'shared': net['shared']
+                    'shared': net['shared'],
+                    'router:external': net['router:external'],
+                    'provider:physical_network': net['provider:physical_network'],
+                    'provider:network_type':  net['provider:network_type']
                 }
             }
             if net['router:external']:
                 if not self.config.migrate.migrate_extnets:
                     continue
-                network_info['network']['router:external'] = \
-                    net['router:external']
-                network_info['network']['provider:physical_network'] = \
-                    net['provider:physical_network']
-                network_info['network']['provider:network_type'] = \
-                    net['provider:network_type']
-                if net['provider:network_type'] == 'vlan':
-                    network_info['network']['provider:segmentation_id'] = \
-                        net['provider:segmentation_id']
```

Comment:
Those lines are not removed. Originally, those keys are only defined for external network. The fix is to define all those key for both external and internal networks.